File: Menu_app/app/src/main/java/com/example/menu_app/Activity/MainActivity.java
package com.example.menu_app.Activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.GridLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.os.Bundle;

import com.example.menu_app.Adapter.dishesRecViewAdapter;
import com.example.menu_app.R;
import com.example.menu_app.Class.dish;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private RecyclerView dishesRecView;
    private final ArrayList<dish> dishes = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        dishesRecView = findViewById(R.id.dishesRecView);

        dishes.add(new dish("Barbecued Spare Ribs with BBQ Sauce", "骨", "骨", "1", 650, 0));
        dishes.add(new dish("Spare Ribs with Chilli and Salt", "椒盐骨", "盐骨", "2",650, 0));
        dishes.add(new dish("Barbecued Spare Ribs with Honey Sauce", "蜜骨", "蜜骨", "3", 650, 0));
        dishes.add(new dish("Sweet & Sour Spare Ribs", "咕咾骨", "古骨", "4", 650, 0));
        dishes.add(new dish("Chicken Wings with Chilli & Salt", "椒盐鸡翅", "盐鸡羽", "8", 650, 0));
        dishes.add(new dish("Crab Claws (4)", "蟹爪", "蟹爪", "21", 360, 0));
        dishes.add(new dish("1/2 Crispy Aromatic Duck", "1/2 香鸭", "1/2 香甲", "28", 880, 0));
        dishes.add(new dish("King Prawns with Green Peppers & Black Bean Sauce", "豉椒大虾", "士大下", "59", 650, 0));
        dishes.add(new dish("Sweet and Sour King Prawns Balls (Cantonese Style)", "中式咕咾大虾", "中古大下", "162", 700, 0));

        dishesRecyclerView();

    }

    private void dishesRecyclerView(){
        dishesRecViewAdapter adapter = new dishesRecViewAdapter(this);
        adapter.setDishes(dishes);

        dishesRecView.setAdapter(adapter);
        dishesRecView.setLayoutManager(new GridLayoutManager(this,3));
    }
}

File: Menu_app/app/src/main/java/com/example/menu_app/Activity/MainActivity.kt
package com.example.menu_app.activity

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.drawerlayout.widget.DrawerLayout
import androidx.navigation.NavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.example.menu.R
import com.google.android.material.navigation.NavigationView

class MainActivity : AppCompatActivity(), NavigationView.OnNavigationItemSelectedListener{

    private lateinit var navController: NavController
    private lateinit var drawerLayout: DrawerLayout
    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var navView: NavigationView
    private lateinit var navHostFragment: NavHostFragment


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        drawerLayout = findViewById(R.id.mainDrawerLayout)
        navHostFragment = supportFragmentManager.findFragmentById(R.id.nav_host_fragment) as NavHostFragment
        navController = navHostFragment.navController

        val topLevelFragments = setOf(
            R.id.mainFragment,
            R.id.databaseFragment
        )

        //appBarConfiguration = AppBarConfiguration(NavController.graph, drawerLayout)
        appBarConfiguration = AppBarConfiguration.Builder(topLevelFragments)
            .setOpenableLayout(drawerLayout)
            .build()

        navView = findViewById(R.id.nav_drawer)
        navView.setNavigationItemSelectedListener(this)
        navView.setupWithNavController(navController)

        setupActionBarWithNavController(navController, appBarConfiguration)
    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId){
            R.id.mainFragment -> {
                navController.navigate(R.id.mainFragment)
            }
            R.id.databaseFragment -> {
                navController.navigate(R.id.databaseFragment)
            }
        }
        drawerLayout.closeDrawers()
        return true
    }

    override fun onSupportNavigateUp(): Boolean {
        navController = navHostFragment.navController
        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
    }
}